Machine I/O Explicit Sides

FrederoxDev opened the pull request:
This PR adds the ability for machines to explicitly allow types/categories on specific sides of a machine

To enable this functionality on a machine add the `fluffyalien_energisticscore:explicit_sides` tag. These changes do not effect any machines without this tag.

To define a machine which takes in water on specifically only the east side, you would need this tag: `fluffyalien_energisticscore:io.type.water.east`

## Explicit Sides Syntax
`fluffyalien_energisticscore:io.{type|category}.<StorageTypeId>.{north|east|south|west|up|down|side}`
**or**
`fluffyalien_energisticscore:io.any.{north|east|south|west|up|down|side}`

## Breaking Changes

This PR replaces `MachineIo` with `MachineSideIo`.

FrederoxDev commented:
Unsure if that run fail is my fault, error seems to be unrelated? Compiles fine for me, all issues found by `npm run check` are unrelated to anything I've done

Fluffyalien1422 commented:
> Unsure if that run fail is my fault, error seems to be unrelated? Compiles fine for me, all issues found by `npm run check` are unrelated to anything I've done

I just reran the job and it succeeded. It seems like it was a GitHub issue.

Fluffyalien1422 commented:
I don't think that the `fluffyalien_energisticscore:explicit_sides.enable_cardinal_rotation` tag is a good idea.

We don't actually know the visual rotation of the block based on the `minecraft:cardinal_direction` state. For example, depending on how the block model is rotated, the `minecraft:rotation` values may be different. A block is also not required to use the `minecraft:cardinal_direction` state for rotation (some vanilla blocks don't even use it).

I think it would be better to just have the developer add the tags in the rotation permutations, then `MachineNetwork.updateWithBlock` can be called if the rotation changes.

FrederoxDev commented:
Changes:
- Removed all rotation features as requested
- Renamed MachineIo to MachineIoSide
- Added support for sides to allow any with. `fluffyalien_energisticscore:io.any.{north|east|south|west|up|down|side}`

FrederoxDev commented:
Just snuck another commit in that fixes a small typo of the project namespace which was causing destroyNetwork to not work
